core/src/main/java/com/example/AutomatedReviewTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.example;
+
+import java.awt.event.*;
+import java.io.*;
+import java.util.*;
+
+import javax.swing.*;
+
+import java.time.*;
+
+/**
+ * The goal of this code is to find out if code analysis will automatically add
+ * commentary to a pull request when it finds problems with the code.
+ */
+final class AutomatedReviewTest extends JFrame implements ActionListener {
+  
+  private static Object lazy;
+  
+  AutomatedReviewTest() {
+    pack(); // should spawn new thread
+    setVisible(true);
+  }
+  
+  static Object getLazy() {
+    if (lazy == null) {
+      lazy = new Object();
+    }
+    
+    return lazy;
+  }
+  
+  synchronized boolean getIsTest(String s) {
+    try {
+      return s.matches("Test");
+    } catch (Exception ex) {
+      throw new RuntimeException();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://sonarcloud.io/organizations/coderanch-corral/rules#rule_key=squid%3AS00112))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.coderanch%3Ablackjack&pullRequest=9&issues=AWZEavo0nzfF1D4J800F&open=AWZEavo0nzfF1D4J800F)

[ISSUE_KEY]: AWZEavo0nzfF1D4J800F